sw/device/lib/testing/usb_testutils_streams.c

```diff
@@ -697,6 +702,81 @@ status_t usb_testutils_streams_init(usb_testutils_streams_ctx_t *ctx,
   return OK_STATUS();
 }
 
+status_t usb_testutils_streams_typed_init(
+    usb_testutils_streams_ctx_t *ctx, uint8_t *cfg, uint16_t len,
+    unsigned nstreams, const usb_testutils_transfer_type_t xfr_types[],
+    uint32_t num_bytes, usbdev_stream_flags_t flags, bool verbose,
+    uint32_t *types) {
+  TRY_CHECK(nstreams <= USBUTILS_STREAMS_MAX);
+
+  // Does the caller require a bitmap of the transfer type(s) collected?
+  if (types) {
+    *types = 0U;
+  }
+
+  // Total length of the configuration descriptor; validate caller buffer.
+  size_t cfg_len = USB_CFG_DSCR_LEN +
+                   nstreams * (USB_INTERFACE_DSCR_LEN + 2 * USB_EP_DSCR_LEN);
+  TRY_CHECK(cfg && cfg_len <= len);
+
+  // Configuration Descriptor header.
+  uint8_t head[CFG_DSCR_LEN_MAX] = {
+      USB_CFG_DSCR_HEAD((uint16_t)cfg_len, (uint8_t)nstreams)};
+  memcpy(cfg, head, USB_CFG_DSCR_LEN);
+  cfg += USB_CFG_DSCR_LEN;
+
+  // Followed by programmatically-generated list of interface descriptors.
+  for (uint8_t id = 0U; id < nstreams; id++) {
+    usb_testutils_transfer_type_t xfr_type = xfr_types[id];
+    // Return to the caller the transfer type of each stream in turn.
+    if (types) {
+      *types |= xfr_type << (id * 2U);
+    }
+
+    uint8_t ep_in = (uint8_t)(id + 1U);
+    uint8_t ep_out = (uint8_t)(id + 1U);
+    TRY(usb_testutils_stream_init(ctx, id, xfr_type, ep_in, ep_out, num_bytes,
+                                  flags, verbose));
+
+    // Isochronous and Interrupt endpoints require a bInterval value of 1.
+    uint8_t bInterval = (xfr_type == kUsbTransferTypeIsochronous ||
```

Comment:
Although this valid C code, I think it would be clearer to explicitely write `(...) ? 1 : 0`.